contracts/RelayerHub.sol

```diff
@@ -8,88 +8,206 @@ import "./System.sol";
 import "./lib/SafeMath.sol";
 
 
-contract RelayerHub is IRelayerHub, System, IParamSubscriber{
-  using SafeMath for uint256;
-
-  uint256 public constant INIT_REQUIRED_DEPOSIT =  1e20;
-  uint256 public constant INIT_DUES =  1e17;
-
-  uint256 public requiredDeposit;
-  uint256 public dues;
-
-  mapping(address =>relayer) relayers;
-  mapping(address =>bool) relayersExistMap;
-
-  struct relayer{
-    uint256 deposit;
-    uint256  dues;
-  }
-
-  modifier notContract() {
-    require(!isContract(msg.sender), "contract is not allowed to be a relayer");
-    _;
-  }
-
-  modifier noProxy() {
-    require(msg.sender == tx.origin, "no proxy is allowed");
-    _;
-  }
-
-  modifier noExist() {
-    require(!relayersExistMap[msg.sender], "relayer already exist");
-    _;
-  }
-
-  modifier exist() {
-    require(relayersExistMap[msg.sender], "relayer do not exist");
-    _;
-  }
-
-  event relayerRegister(address _relayer);
-  event relayerUnRegister(address _relayer);
-  event paramChange(string key, bytes value);
-
-
-  function init() external onlyNotInit{
-    requiredDeposit = INIT_REQUIRED_DEPOSIT;
-    dues = INIT_DUES;
-    alreadyInit = true;
-  }
-  
-  function register() external payable noExist onlyInit notContract noProxy{
-    revert("register suspended");
-  }
-  
-
-  function  unregister() external exist onlyInit{
-    relayer memory r = relayers[msg.sender];
-    msg.sender.transfer(r.deposit.sub(r.dues));
-    address payable systemPayable = address(uint160(SYSTEM_REWARD_ADDR));
-    systemPayable.transfer(r.dues);
-    delete relayersExistMap[msg.sender];
-    delete relayers[msg.sender];
-    emit relayerUnRegister(msg.sender);
-  }
-
-  /*********************** Param update ********************************/
-  function updateParam(string calldata key, bytes calldata value) external override onlyInit onlyGov{
-    if (Memory.compareStrings(key,"requiredDeposit")) {
-      require(value.length == 32, "length of requiredDeposit mismatch");
-      uint256 newRequiredDeposit = BytesToTypes.bytesToUint256(32, value);
-      require(newRequiredDeposit > 1 && newRequiredDeposit <= 1e21 && newRequiredDeposit > dues, "the requiredDeposit out of range");
-      requiredDeposit = newRequiredDeposit;
-    } else if (Memory.compareStrings(key,"dues")) {
-      require(value.length == 32, "length of dues mismatch");
-      uint256 newDues = BytesToTypes.bytesToUint256(32, value);
-      require(newDues > 0 && newDues < requiredDeposit, "the dues out of range");
-      dues = newDues;
-    } else {
-      require(false, "unknown param");
-    }
-    emit paramChange(key, value);
-  }
-
-  function isRelayer(address sender) external override view returns (bool) {
-    return relayersExistMap[sender];
-  }
+contract RelayerHub is IRelayerHub, System, IParamSubscriber {
+    using SafeMath for uint256;
+
+    bool public alreadyUpdate;
```

Comment:
The whole storage structure will break if you add a new var here. For any upgrade smart contract, you can only append new variables, the order matters. (This is because of the smart contract storage layout)

Reply:
`alreadyUpdate` is a too generic name, since we upgrade this contract in the future, by then if we need add another `alreadyUpdate`, that will be confusing.